chezmoi crash while applying dotfiles on github action

## What exactly are you trying to do?

I'm trying to make a CI to test my dotfiles (for syntax, chezmoi tmpl,...). But when it comes to "chezmoi apply" step, it fails.

> [!NOTE]
>
> I don't know if this idea is realistic or not. I would be happy if you could suggest me any additions and modifications in this flow

## What have you tried so far?

Wrote a github action workflow file

```yaml
name: Test Install

on:
  push:
    branches:
      - main

env:
  BIN_PATH: ${{ github.workspace }}/bin
  INIT_ARGS: >
    --depth=1
    --promptDefaults
    --no-tty
    --color=false
  APPLY_ARGS: >
    --exclude='encrypted,externals,scripts'
    --parent-dirs
    --no-tty
    --color=false
  DOCTOR_ARGS: >
    --keep-going 
    --no-tty
    --color=false

jobs:
  test:
    strategy:
      fail-fast: true
      matrix:
        os:
          - windows
          - ubuntu
          - macos
    runs-on: ${{ matrix.os }}-latest
    name: ${{ matrix.os }}
    steps:
      - name: Install chezmoi
        shell: bash
        run: bash <(curl -fsLS get.chezmoi.io) -b ${{ env.BIN_PATH }}

      - name: Init chezmoi
        shell: bash
        run: ${{ env.BIN_PATH }}/chezmoi init ${{ github.repository_owner }} ${{ env.INIT_ARGS }}

      - name: Chezmoi apply
        shell: bash
        run: ${{ env.BIN_PATH }}/chezmoi apply ${{ env.APPLY_ARGS }}

      - name: Chezmoi doctor
        shell: bash
        run: ${{ env.BIN_PATH }}/chezmoi doctor ${{ env.DOCTOR_ARGS }}

      - name: Check all template
        shell: bash
        run: |
          tmpls=$(find . -type f -name "*.tmpl" ! -name ".chezmoi.toml.tmpl")
          for tmpl in "${tmpls[@]}"; do
            chezmoi execute-template < "$tmpl" &> /dev/null
            if [ $? -ne 0 ]; then
              echo "$tmpl: Tmpl error" 1>&2
            fi
          done
```

> Workflow run status <https://github.com/KevinNitroG/dotfiles/actions/runs/13202682127>

And I got this error on ubuntu host while running "chezmoi apply"

```log
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xa567ea]
goroutine 1 [running]:
github.com/twpayne/chezmoi/v2/internal/chezmoi.(*EntryTypeSet).ContainsSourceStateEntry(0xc0026761c8, {0x0, 0x0})
	/home/runner/work/chezmoi/chezmoi/internal/chezmoi/entrytypeset.go:126 +0x2a
github.com/twpayne/chezmoi/v2/internal/chezmoi.(*EntryTypeFilter).IncludeSourceStateEntry(0xc00298c4e0, {0x0, 0x0})
	/home/runner/work/chezmoi/chezmoi/internal/chezmoi/entrytypefilter.go:34 +0x28
github.com/twpayne/chezmoi/v2/internal/chezmoi.(*SourceState).Apply(0xc0029c56c0, {0x20a8360, 0xc0005d6e40}, {0x20a8360, 0xc0005d6e40}, {0x209e9f0, 0xc0029a8990}, {0xc00004a095, 0xc}, {{0xc0032c0270, ...}}, ...)
	/home/runner/work/chezmoi/chezmoi/internal/chezmoi/sourcestate.go:708 +0xc6
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).applyArgs(0xc0029be008, {0x2094ab8, 0x394fe80}, {0x20a8360, 0xc0005d6e40}, {0xc00004a095, 0xc}, {0xc0005d6d00, 0x0, 0x4}, ...)
	/home/runner/work/chezmoi/chezmoi/internal/cmd/config.go:695 +0x67b
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).runApplyCmd(0xc0029be008, 0xc001f9f880?, {0xc0005d6d00, 0x0, 0x4})
	/home/runner/work/chezmoi/chezmoi/internal/cmd/applycmd.go:41 +0x165
github.com/spf[13](https://github.com/KevinNitroG/dotfiles/actions/runs/13202682127/job/36858337933#step:4:14)/cobra.(*Command).execute(0xc0029c8308, {0xc0005d6cc0, 0x4, 0x4})
	/home/runner/.cache/go-build/github.com/spf13/cobra@v1.8.1/command.go:985 +0xaaa
github.com/spf13/cobra.(*Command).ExecuteC(0xc002474f08)
	/home/runner/.cache/go-build/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/.cache/go-build/github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).execute(0xc0029be008, {0xc000040250, 0x5, 0x5})
	/home/runner/work/chezmoi/chezmoi/internal/cmd/config.go:1287 +0xbb
github.com/twpayne/chezmoi/v2/internal/cmd.runMain({{0x207a290, 0x6}, {0x2090c40, 0x28}, {0x2080da0, 0x[14](https://github.com/KevinNitroG/dotfiles/actions/runs/13202682127/job/36858337933#step:4:15)}, {0x207b380, 0xa}}, {0xc000040250, 0x5, ...})
	/home/runner/work/chezmoi/chezmoi/internal/cmd/cmd.go:347 +0x22a
github.com/twpayne/chezmoi/v2/internal/cmd.Main({{0x207a290, 0x6}, {0x2090c40, 0x28}, {0x2080da0, 0x14}, {0x207b380, 0xa}}, {0xc000040250, 0x5, ...})
	/home/runner/work/chezmoi/chezmoi/internal/cmd/cmd.go:99 +0xa5
main.main()
	/home/runner/work/chezmoi/chezmoi/main.go:30 +0x107
```

Also on MacOS runner I got this error. I though it might be fixed with `--parent-dirs` but it isn't

```log
chezmoi: .config/kitty/kitty.conf: stat /Users/runner/.config/kitty: no such file or directory
```

## Where else have you checked for solutions?

* [x] I have read [chezmoi's user guide](https://chezmoi.io/user-guide/command-overview/), and not found the answer.
* [x] I have searched [chezmoi's reference guide](https://chezmoi.io/reference/), and not found the answer.
* [ ] Other, please give details.

--- a/internal/chezmoi/sourcestate.go
+++ b/internal/chezmoi/sourcestate.go
@@ -704,6 +704,9 @@ func (s *SourceState) Apply(
 	options ApplyOptions,
 ) error {
 	sourceStateEntry := s.root.get(targetRelPath)
+	if sourceStateEntry == nil {
+		return nil
+	}
 
 	if !options.Filter.IncludeSourceStateEntry(sourceStateEntry) {
 		return nil

